src/main/resources/public/ts/directives/delete-world/delete-world.directive.ts

```diff
@@ -41,29 +39,19 @@ class Controller implements ng.IController, IViewModel {
     }
 
     async deleteWorld(): Promise<void> {
-        minetestService.delete(this.world)
-            .then(() => {
-                toasts.confirm('minetest.world.delete.confirm');
-                this.closeDeleteLightbox();
-                this.$scope.$parent.$eval(this.$scope['vm']['onDeleteWorld'])(this.world);
-            })
-            .catch(e => {
-                toasts.warning('minetest.world.delete.error');
-                console.error(e);
-                this.closeDeleteLightbox();
-            })
-    }
-
-    async deleteImportWorld(): Promise<void> {
-        minetestService.deleteImportWorld(this.world)
-            .then(() => {
-                toasts.confirm('minetest.world.delete.confirm');
-                this.closeDeleteLightbox();
-                this.$scope.$parent.$eval(this.$scope['vm']['onDeleteWorld'])(this.world);
-            }).catch(() => {
-            toasts.warning('minetest.world.delete.error');
-            this.closeDeleteLightbox();
-        })
+        this.worlds.all.forEach((world: IWorld) => {
+            (world['isExternal'])? minetestService.deleteImportWorld(world) : minetestService.delete(world)
+                .then(() => {
+                    toasts.confirm('minetest.world.delete.confirm');
+                    this.closeDeleteLightbox();
+                    this.$scope.$parent.$eval(this.$scope['vm']['onDeleteWorld'])(this.worlds);
+                })
+                .catch(e => {
+                    toasts.warning('minetest.world.delete.error');
+                    console.error(e);
+                    this.closeDeleteLightbox();
+                })
```

Comment:
```suggestion
            let response: Promise<AxiosResponse> = (world['isExternal']) ? minetestService.deleteImportWorld(world) : minetestService.delete(world);
            response
                .then(() => {
                    toasts.confirm('minetest.world.delete.confirm');
                    this.closeDeleteLightbox();
                    this.$scope.$parent.$eval(this.$scope['vm']['onDeleteWorld'])(this.worlds);
                })
                .catch(e => {
                    toasts.warning('minetest.world.delete.error');
                    console.error(e);
                    this.closeDeleteLightbox();
                })
```
Ton then/catch ne s'applique que a minetestService.delete et pas a minetestService.deleteImportWorld